stellargraph/core/graph.py

```diff
@@ -303,19 +303,31 @@ def nodes(self) -> Iterable[Any]:
 
         return self._nodes.ids.pandas_index
 
-    def edges(self, include_edge_type=False) -> Iterable[Any]:
+    def edges(
+        self, include_edge_type=False, include_edge_weight=False
+    ) -> Iterable[Any]:
         """
         Obtains the collection of edges in the graph.
 
         Args:
-            include_edge_type (bool): A flag that indicates whether to return edge triples
+            include_edge_type (bool): A flag that indicates whether to return edge types
             of format (node 1, node 2, edge type) or edge pairs of format (node 1, node 2).
+            include_edge_weight (bool): A flag that indicates whether to return edge weights.
+            Weights are returned in a separate list.
 
         Returns:
-            The graph edges.
+            The graph edges. If edge weights are included then a tuple of (edges, weights)
         """
         if self._graph is not None:
-            return self._graph.edges(include_edge_type)
+            return self._graph.edges(
+                include_edge_weight=include_edge_weight,
+                include_edge_type=include_edge_type,
+            )
+
+        if include_edge_weight:
```

Comment:
Given how you've implemented the `graph_networkx` version, I think this is possible in almost the same way, and so should be done here:

```python
if include_edge_type:
    edges = list(zip(...))
else:
    edges = list(zip(...))

if include_edge_weight:
    return edges, self._edges.weights

return edges
```